instrumentation/failsafe-3.0/library/src/main/java/io/opentelemetry/instrumentation/failsafe/v3_0/FailsafeTelemetry.java

```diff
@@ -70,4 +76,41 @@ public <R> CircuitBreaker<R> createCircuitBreaker(
         .onClose(buildInstrumentedCloseListener(userConfig, stateChangesCounter, attributes))
         .build();
   }
+
+  /**
+   * Returns an instrumented {@link RetryPolicy} by given values.
+   *
+   * @param delegate user configured {@link RetryPolicy} to be instrumented
+   * @param retryPolicyName identifier of given {@link RetryPolicy}
+   * @param <R> {@link RetryPolicy}'s result type
+   * @return instrumented {@link RetryPolicy}
+   */
+  public <R> RetryPolicy<R> createRetryPolicy(RetryPolicy<R> delegate, String retryPolicyName) {
+    RetryPolicyConfig<R> userConfig = delegate.getConfig();
+    Meter meter = openTelemetry.getMeter(INSTRUMENTATION_NAME);
+    LongCounter executionCounter =
+        meter
+            .counterBuilder("failsafe.retry_policy.execution.count")
+            .setDescription(
+                "Count of execution events processed by the retry policy. "
+                    + "Each event represents one complete execution flow (initial attempt + any retries). "
+                    + "This metric does not count individual retry attempts - it counts each time the policy is invoked.")
```

Comment:
just an idea, but perhaps we could minimize the length of this description a bit by encoding some of this information in the unit?

```suggestion
                "Count of execution events processed by the retry policy.")
            .setUnit("{policy_invocation}")
         
```